portfolio/src/main/java/com/google/sps/servlets/ChartServlet.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.LinkedHashMap;
+import java.util.Scanner;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/chart-data")
+public class ChartServlet extends HttpServlet {
+
+  // Stores data in the form "country_name: rice_produced."
+  private LinkedHashMap<String, Double> countryMap = new LinkedHashMap<>();
+
+  @Override
+  public void init() {
+    Scanner scanner = new Scanner(getServletContext().getResourceAsStream("/WEB-INF/edible_food_2011.csv"));
+
+    String headerLine1 = scanner.nextLine();
+    String headerLine2 = scanner.nextLine();
+    String headerLine3 = scanner.nextLine();
```

Comment:
Are these header lines determinate?  Can you use a method in the Scanner class to figure out when the real content starts?
https://docs.oracle.com/javase/7/docs/api/java/util/Scanner.html

I say this because 1) you're creating 3 headerLine variables which don't seem to be used and 2) this way of "skipping" the first 3 lines is error-prone if the file format changes for some reason.

Reply:
thanks for the link! I'm editing the while loop to check internally if the line contains valid data.